addons/point_of_sale/models/pos_order.py

```diff
@@ -1007,6 +1008,17 @@ def _order_line_fields(self, line, session_id=None):
     tax_ids_after_fiscal_position = fields.Many2many('account.tax', compute='_get_tax_ids_after_fiscal_position', string='Taxes to Apply')
     pack_lot_ids = fields.One2many('pos.pack.operation.lot', 'pos_order_line_id', string='Lot/serial Number')
     product_uom_id = fields.Many2one('uom.uom', string='Product UoM', related='product_id.uom_id')
+    income_account_id = fields.Many2one('account.account', string='Income account', compute='_compute_income_account')
+
+    @api.depends('product_id', 'order_id.fiscal_position_id')
+    def _compute_income_account(self):
```

Comment:
can't you just call the function 'def get_product_accounts(self, fiscal_pos=None)' defined on product and pass it the fiscal_position as parameter?